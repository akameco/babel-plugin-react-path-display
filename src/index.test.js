// @flow
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
    parserOpts: {
      plugins: ['classProperties', 'jsx'],
    },
    babelrc: false,
  },
  tests: [
    {
      title: 'ファイルパスに従って、displayNameを作成する',
      code: `
    import React from 'react'

    export default class Link extends React.Component {
      render() {
        return <div>hello world</div>
      }
    }
    `,
    },
    {
      title: 'displayNameがすでにある場合は変換しない',
      code: `
    import React from 'react'

    export default class Link extends React.Component {
      static displayName = 'Link'
      render() {
        return <div>hello world</div>
      }
    }
    `,
    },
    {
      title: 'SFCの場合は変換しない',
      code: `
    import React from 'react'

    function Link() {
      return <div>hello</div>
    }

    export default Link
    `,
    },
    {
      title: 'React以外のクラスは変換しない',
      code: `
    export default class Link {
      render() {
        return <div>hello world</div>
      }
    }
    `,
    },
    {
      title: 'displayName以外のstatic class propertyがある場合',
      code: `
    import React from 'react'

    export default class Link extends React.Component {
      static hello = 'hello'
      render() {
        return <div>hello world</div>
      }
    }
    `,
    },
  ],
})
