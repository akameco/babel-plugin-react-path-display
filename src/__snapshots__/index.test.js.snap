// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React以外のクラスは変換しない 1`] = `
"
export default class Link {
  render() {
    return <div>hello world</div>
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

export default class Link {
  render() {
    return <div>hello world</div>;
  }
}
"
`;

exports[`SFCの場合は変換しない 1`] = `
"
import React from 'react'

function Link() {
  return <div>hello</div>
}

export default Link

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

function Link() {
  return <div>hello</div>;
}

export default Link;
"
`;

exports[`displayNameがすでにある場合は変換しない 1`] = `
"
import React from 'react'

export default class Link extends React.Component {
  static displayName = 'Link'
  render() {
    return <div>hello world</div>
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default class Link extends React.Component {
  static displayName = 'Link';
  render() {
    return <div>hello world</div>;
  }
}
"
`;

exports[`displayName以外のstatic class propertyがある場合 1`] = `
"
import React from 'react'

export default class Link extends React.Component {
  static hello = 'hello'
  render() {
    return <div>hello world</div>
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default class Link extends React.Component {
  static hello = 'hello';
  render() {
    return <div>hello world</div>;
  }
  static displayName = 'src.index.test.Link';
}
"
`;

exports[`ファイルパスに従って、displayNameを作成する 1`] = `
"
import React from 'react'

export default class Link extends React.Component {
  render() {
    return <div>hello world</div>
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

import React from 'react';

export default class Link extends React.Component {
  render() {
    return <div>hello world</div>;
  }
  static displayName = 'src.index.test.Link';
}
"
`;
